### PR TITLE
Update SMCP template for 3scale adapter

### PR DESCRIPTION
**Pull request by philipgough**
@knrc - this is intended to fix https://issues.jboss.org/browse/MAISTRA-1004 and issues discussed in https://issues.jboss.org/browse/OSSM-106  but not sure if it covers everything and not sure what the target branch should be at this point.

### CODE REVIEW

File: deploy/smcp-templates/servicemesh [hunk 1]
```diff
@@ -31,4 +31,4 @@ spec:
       image: prometheus-rhel8
   threeScale:
     hub: registry.redhat.io/openshift-service-mesh
-    tag: 1.0.0
```
Comment (brian-avery) [inline deploy/smcp-templates/servicemesh:34]: Did you mean to remove the tag?

**Comment by brian-avery** [comment]
This should probably be merged into maistra-1.0

File: deploy/smcp-templates/servicemesh (hunk 1 above)
Comment (philipgough) [reply deploy/smcp-templates/servicemesh:34]: I did at the time because I was considering always pulling latest but after some thought I reverted that

**Comment by knrc** [comment]
@PhilipGough Could you please rebase this on the 1.0 branch?  It looks as if your source conflicts.

File: deploy/smcp-templates/servicemesh (hunk 1 above)
Comment (knrc) [reply deploy/smcp-templates/servicemesh:34]: We wouldn't be using latest in a production environment so should include the explicit tag or a floating tag

**Comment by philipgough** [comment]
@knrc done. I changed the base via the UI and was in the process of cleaning the mess when you commented 👍

File: deploy/smcp-templates/servicemesh (hunk 1 above)
Comment (philipgough) [reply deploy/smcp-templates/servicemesh:34]: Yep, I came to that realisation and reverted the change

**Review by brian-avery — APPROVED** [review]

**Review by dgn — APPROVED** [review]

**Pull request merged**